<a href="https://colab.research.google.com/github/Heeun93/2021-36845_Homework/blob/main/220602_project_2021_36845_Heeun_Kwon.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##데이터 다운 및 환경설정
Google colab 기반으로 데이터 파일들을 불러오고 저장하기 위해서 Google Drive를 연결함

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
!git clone https://github.com/hyeshik/colab-biolab.git
!cd colab-biolab && bash tools/setup.sh
exec(open('colab-biolab/tools/activate_conda.py').read())

Cloning into 'colab-biolab'...
remote: Enumerating objects: 69, done.
remote: Counting objects: 100% (69/69), done.
remote: Compressing objects: 100% (43/43), done.
remote: Total 69 (delta 22), reused 54 (delta 13), pack-reused 0
Unpacking objects: 100% (69/69), done.
./
./root/
./root/.vimrc
./root/.condarc
./root/.tmux.conf
./root/.profile
./root/.bashrc.biolab
./root/.bin.priority/
./root/.bin.priority/pip
./root/.bin.priority/pip3
./root/.bin.priority/pip2
--2022-06-06 16:31:21--  https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.131.3, 104.16.130.3, 2606:4700::6810:8303, ...
Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.131.3|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 76607678 (73M) [application/x-sh]
Saving to: ‘miniconda3.sh’

miniconda3.sh       100%[===================>]  73.06M   233MB/s    in 0.3s    

2022-06-06 16:31:21 (233 MB/s) - ‘miniconda3.sh’ save

지난 과제 때 받았던 시퀀싱 데이터 (binfo1-datapack1.tar)를 사용. 혹시 모르니 파일 위치 확인 및 MD5 체크섬 다시 확인.

```
140aaf30bcb9276cc716f8699f04ddd6  CLIP-35L33G.bam
f1b3336ed7e2f97d562dcc71641251bd  CLIP-35L33G.bam.bai
328883a73d507eafbf5b60bd6b906201  RNA-control.bam
02073818e2f398a73c3b76e5169de1ca  RNA-control.bam.bai
b09550d09d6c2a4ce27f0226f426fdb1  RNA-siLin28a.bam
fef112c727244060ea62d3f2564a07f6  RNA-siLin28a.bam.bai
28bbd0c47d725669340c784f1b772c01  RNA-siLuc.bam
43590fdc4d81905c0432e0d1cb8cfd5b  RNA-siLuc.bam.bai
5c08a9297307bc83259e658c4474f0cc  RPF-siLin28a.bam
a1bb3e29be412dfd7fd8d16b1b1acc4c  RPF-siLin28a.bam.bai
f2eebf50943024d0116c9cd3e744c707  RPF-siLuc.bam
dc24f69e8f571fc8be30f28ce5b84fcd  RPF-siLuc.bam.bai
```

In [ ]:
!md5sum drive/MyDrive/binfo1-datapack1/*

140aaf30bcb9276cc716f8699f04ddd6  drive/MyDrive/binfo1-datapack1/CLIP-35L33G.bam
f1b3336ed7e2f97d562dcc71641251bd  drive/MyDrive/binfo1-datapack1/CLIP-35L33G.bam.bai
9f43477870337dfddbc154809a2a3ec8  drive/MyDrive/binfo1-datapack1/CLIP-let7d.bam
8f0013aa795510f597e241c1b69e6341  drive/MyDrive/binfo1-datapack1/CLIP-let7d-gene.pileup
8f0013aa795510f597e241c1b69e6341  drive/MyDrive/binfo1-datapack1/CLIP-let7d.pileup
d5d8996d383d994fe1916bf239f91964  drive/MyDrive/binfo1-datapack1/CLIP-let7f-1.bam
1c34f27bcc5f407e54d8112938bc9d2c  drive/MyDrive/binfo1-datapack1/CLIP-let7f-1-gene.pileup
9b172b22ce452b8f7ef1b02dcf61516b  drive/MyDrive/binfo1-datapack1/CLIP-let7f-1.pileup
0d33bf4b3a5ce22ba13a62323efd6883  drive/MyDrive/binfo1-datapack1/CLIP-let7g.bam
4ea14471c5d4bf9ab1fd91b9a5b2193c  drive/MyDrive/binfo1-datapack1/CLIP-let7g-gene.pileup
a53f66a5575d4ac2d3b4c44af576d44d  drive/MyDrive/binfo1-datapack1/CLIP-let7g.pileup
md5sum: drive/MyDrive/binfo1-datapack1/colab-biolab: Is a directory
md5sum:

실습에 필요한 프로그램들 몇 가지를 추가로 설치.

In [3]:
!conda install -y bedtools bioawk

Solving environment: | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | done


==> WARNING: A newer version of conda exists. <==
  current version: 4.12.0
  latest version: 4.13.0

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /root/conda/envs/lab

  added / updated specs:
    - bedtools
    - bioawk


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    bioawk-1.0                 |       h7132678_7         196 KB  bioconda
    ------------------------------------------------------------
                                           Total:         196 KB

The following NEW packages will be INSTALLED:

  bioawk             bioconda/linux-64::bioawk-1.0-h7132678_7



bioawk-1.0           | 196 KB    | : 100% 1.0/1 [00:00<00:00,  6.42it/s]   

## Fig S2A 재현해보기 - error가 많이 나오는 부분의 unique sequence들의 각각 개수를 세서 genome browser로 표현하고, 점수 트랙을 같이 표시하기

참고 논문 ([LIN28A Is a Suppressor of ER-Associated Translation in Embryonic Stem Cells](https://www.cell.com/cell/fulltext/S0092-8674(12)01234-2?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867412012342%3Fshowall%3Dtrue))의 Fig S2A처럼, 오류율을 분석해서 단백질이 앉아 있었던 자리를 분석.

--------------------------------

데이터는 지난 번과 동일하기 때문에, 미리 다운받은 데이터를 사용. 

In [4]:
%cd /content/drive/MyDrive/binfo1-datapack1/

/content/drive/MyDrive/binfo1-datapack1


Mirlet7d 부분을 레퍼런스 mm9 기준으로 솎아냄

In [ ]:
!grep -i mirlet7d gencode.gtf

chr13	ENSEMBL	gene	48689488	48689590	.	-	.	gene_id "ENSMUSG00000065453.3"; gene_type "miRNA"; gene_name "Mirlet7d"; level 3; mgi_id "MGI:2676796";
chr13	ENSEMBL	transcript	48689488	48689590	.	-	.	gene_id "ENSMUSG00000065453.3"; transcript_id "ENSMUST00000083519.3"; gene_type "miRNA"; gene_name "Mirlet7d"; transcript_type "miRNA"; transcript_name "Mirlet7d-201"; level 3; transcript_support_level "NA"; mgi_id "MGI:2676796"; tag "basic";
chr13	ENSEMBL	exon	48689488	48689590	.	-	.	gene_id "ENSMUSG00000065453.3"; transcript_id "ENSMUST00000083519.3"; gene_type "miRNA"; gene_name "Mirlet7d"; transcript_type "miRNA"; transcript_name "Mirlet7d-201"; exon_number 1; exon_id "ENSMUSE00000522678.2"; level 3; transcript_support_level "NA"; mgi_id "MGI:2676796"; tag "basic";


:위의 결과를 반영하여 chromosome number와 범위를 수정.

In [ ]:
!samtools view -b -o CLIP-let7d.bam CLIP-35L33G.bam chr13:48689488-48689590
!samtools view CLIP-let7d.bam | wc -l

187


Fig S2A를 재현해야 하기 때문에, 
1. 솎아낸 CLIP-let7d.bam 안의 read alignments들을 [UCSC Genome Browser](http://genome.ucsc.edu/cgi-bin/hgTracks?db=mm39&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr9%3A106056039-106056126)에 submit하여 병렬로 비교해야 하고,
2. base별 Shannon entropy를 계산해야 함.

##1. CLIP-let7d.bam 안의 read alignments들을 [UCSC Genome Browser](http://genome.ucsc.edu/cgi-bin/hgTracks?db=mm39&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr9%3A106056039-106056126)에 submit

UCSC Genome Browser에서 bam 파일을 직접 custom track으로 submit할 순 없음. 따라서:

- samtools를 이용해 bam file을 sort하고 index file (bam.bai) 생성
- Hosting server에 올림
- [BAM Track Format](https://genomequebec.mcgill.ca/compgen/browser-VPR/goldenPath/help/bam.html) 페이지의 안내에 따라 submit

우선 samtools를 이용해 bam file을 sort하고 index file (bam.bai) 생성.

In [6]:
!samtools sort CLIP-let7d.bam -o CLIP-let7d_sorted.bam

In [7]:
!samtools index CLIP-let7d_sorted.bam

Hosting server는 Github를 이용할 수도 있으나, 편의를 위해 [CyVerse](http://www.cyverse.org/)에 가입하여 [Discovery Environment](https://de.cyverse.org/)에 파일을 업로드한 뒤 호스팅.

[BAM Track Format](https://genomequebec.mcgill.ca/compgen/browser-VPR/goldenPath/help/bam.html) 페이지의 안내에 따라 submit한 후, read alignment가 나오는 것을 확인.

##2. Base별 Shannon entropy를 계산

각 자리에 mapping된 read들의 분포를 요약하기 위해 samtools mpileup을 이용.

In [ ]:
!samtools mpileup CLIP-let7d.bam > CLIP-let7d.pileup
!wc -l CLIP-let7d.pileup

[mpileup] 1 samples in 1 input files
87 CLIP-let7d.pileup


In [ ]:
!head CLIP-let7d.pileup

chr13	48689488	N	24	^Ga^Ia^Ha^Ia^Ha^Ia^Ia^Ia^Ga^Ia^Ia^Ia^Ia^Ia^Ha^Ia^Ga^Ga^Ga^Ia^Ea^Da^Ha^Ga	BIHGGIHIEIIIGIHIGEGH5BDD
chr13	48689489	N	53	aaaaaaaaaaaaaaaaaaaaaaaaa^Ia^Ha^Ia^Ga^Ga^Ha^Ia^Ia^Ga^Ga^Ia^Ia^Ha^Ia^Ia^Ha^Ha^Da^Ia^Ha^Ia^Ia^Ia^Ia^Ia^Ha^Ga^Ga	GIHIGIIIEIIIIIHIG>EB4@5HDIHIBEGIID?IIGII@H>IGHIIIGHDC
chr13	48689490	N	69	ttttttttttttttttttttttttttttttttttttttttttttttttttttt^Gt^Gt^Ht^Gt^It^It^Ht^It^Gt^Gt^Ht^It^It^It^Ht^It	GIHIGHIFFIIIIIFHGEDG;2>FDIHIDGBIIDBIIDIIAH>IBGIIIGHBG=BG=IIGIDGHFIIHI
chr13	48689491	N	105	aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa^Ia^Ia^Ha^Ga^Ia^Ga^Ia^Ga^Ha^Ha^Ha^Ga^Ia^Ha^Ha^Ia^@a^Ia^Ia^Ia^Ga^Ha^Ha^Ga^Ha^Ga^Ia^Ha^Ga^Ia^Ga^Ha^Ga^Ia^Ia^Ga	EIGHEHHIEIIIIIHIGCBH:?>HGGHI>GBIIGBIIGIHBB;IHIIIIGHDC3=G:IGGIDEHIFIGHII=GIDHEGHHEIGGI7III9=HGBAIGBIGFG>I2
chr13	48689492	N	105	aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa	GIGIEIEI7IIIGIGIG;?F:D@G7GHIAGAIIG3IIGIHDB2IHIIIIDDGC1<GCIIHIGCHIIIEIIIGCI

In [ ]:
!awk '$2 >= 48689488 && $2 <= 48689590 { print $0; }' CLIP-let7d.pileup > CLIP-let7d-gene.pileup
!tail CLIP-let7d-gene.pileup

chr13	48689565	N	6	cccccc	>IGIIH
chr13	48689566	N	6	tttttt	:IGFIH
chr13	48689567	N	6	aaaaaa	8IFIFI
chr13	48689568	N	6	cccccc	?IGIII
chr13	48689569	N	6	tttttt	;IGIII
chr13	48689570	N	5	aaaaa	EDECD
chr13	48689571	N	6	cccccc	:GGIGG
chr13	48689572	N	6	cccccc	:GGIGG
chr13	48689573	N	6	tttttt	5GGIGG
chr13	48689574	N	6	c$c$c$c$c$c$	;GGIGG


In [ ]:
!ls

CLIP-35L33G.bam				 gencode-start.gtf
CLIP-35L33G.bam.bai			 Mirlet7d.txt
CLIP-let7d.bam				 Mirlet7f-1.txt
CLIP-let7d-gene.pileup			 Mirlet7g.txt
CLIP-let7d.pileup			 read-counts.txt
CLIP-let7f-1.bam			 read-counts.txt.summary
CLIP-let7f-1-gene.pileup		 RNA-control.bam
CLIP-let7f-1.pileup			 RNA-control.bam.bai
CLIP-let7g.bam				 RNA-siLin28a.bam
CLIP-let7g-gene.pileup			 RNA-siLin28a.bam.bai
CLIP-let7g.pileup			 RNA-siLuc.bam
colab-biolab				 RNA-siLuc.bam.bai
entropy					 RPF-siLin28a.bam
filtered-RPF-siLuc.bam			 RPF-siLin28a.bam.bai
fivepcounts-filtered-RPF-siLuc.txt	 RPF-siLuc.bam
fivepcounts-RPF-siLuc.bed		 RPF-siLuc.bam.bai
gencode-exons-containing-startcodon.bed  test2.bam
gencode.gtf				 test.txt
gencode-plusexon.gtf


pandas에 불러와서 분석

In [ ]:
import pandas as pd
dpileup = pd.read_csv('CLIP-let7d-gene.pileup', sep='\t', names=['chrom', 'pos', '_ref', 'count', 'basereads', 'quals'])
dpileup.tail()

chrom       pos _ref  count     basereads   quals
82  chr13  48689570    N      5         aaaaa   EDECD
83  chr13  48689571    N      6        cccccc  :GGIGG
84  chr13  48689572    N      6        cccccc  :GGIGG
85  chr13  48689573    N      6        tttttt  5GGIGG
86  chr13  48689574    N      6  c$c$c$c$c$c$  ;GGIGG

위에서 구한 pileup에서 match와 substitution 이용

In [ ]:
import re
toremove = re.compile('[<>$*#^]')
dpileup['matches'] = dpileup['basereads'].apply(lambda x: toremove.sub('', x))

In [ ]:
dpileup[['chrom', 'pos', 'matches']]

chrom       pos                                            matches
0   chr13  48689488   GaIaHaIaHaIaIaIaGaIaIaIaIaIaHaIaGaGaGaIaEaDaHaGa
1   chr13  48689489  aaaaaaaaaaaaaaaaaaaaaaaaaIaHaIaGaGaHaIaIaGaGaI...
2   chr13  48689490  tttttttttttttttttttttttttttttttttttttttttttttt...
3   chr13  48689491  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...
4   chr13  48689492  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...
..    ...       ...                                                ...
82  chr13  48689570                                              aaaaa
83  chr13  48689571                                             cccccc
84  chr13  48689572                                             cccccc
85  chr13  48689573                                             tttttt
86  chr13  48689574                                             cccccc

[87 rows x 3 columns]

In [ ]:
dpileup[pileup['pos'] == 106056094].iloc[0]['matches']

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  """Entry point for launching an IPython kernel.


'ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc'

In [ ]:
dpileup2 = dpileup[['chrom', 'pos', 'matches']]

In [ ]:
dpileup3 = dpileup2['matches']

1. 각 position별로 base수를 셈.

In [ ]:
dpileup2['base'] = dpileup3.str.len()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [ ]:
dpileup2

chrom       pos                                            matches  base
0   chr13  48689488   GaIaHaIaHaIaIaIaGaIaIaIaIaIaHaIaGaGaGaIaEaDaHaGa    48
1   chr13  48689489  aaaaaaaaaaaaaaaaaaaaaaaaaIaHaIaGaGaHaIaIaGaGaI...    81
2   chr13  48689490  tttttttttttttttttttttttttttttttttttttttttttttt...    85
3   chr13  48689491  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...   141
4   chr13  48689492  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...   105
..    ...       ...                                                ...   ...
82  chr13  48689570                                              aaaaa     5
83  chr13  48689571                                             cccccc     6
84  chr13  48689572                                             cccccc     6
85  chr13  48689573                                             tttttt     6
86  chr13  48689574                                             cccccc     6

[87 rows x 4 columns]

2. Shannon entropy 계산.

In [ ]:
import math
from collections import Counter

def entropy(s):
	p, lns = Counter(s), float(len(s))
	return -sum( count/lns * math.log(count/lns, 2) for count in p.values())

In [ ]:
count = ['1','2']
dent = pd.Series(count)
print(dent)
print(dent[1])

0    1
1    2
dtype: object
2


In [ ]:
count = ['1','2']
ent = pd.Series(count)
print(ent)
print(ent[1])

0    1
1    2
dtype: object
2


In [ ]:
i = 0
while i <= 86:
  dent[i] = entropy(dpileup3[i])
  i += 1

In [ ]:
dpileup4 = dpileup2
dpileup4['Shannon_entropy'] = dent
print(dpileup4)

    chrom       pos                                            matches  base  \
0   chr13  48689488   GaIaHaIaHaIaIaIaGaIaIaIaIaIaHaIaGaGaGaIaEaDaHaGa    48   
1   chr13  48689489  aaaaaaaaaaaaaaaaaaaaaaaaaIaHaIaGaGaHaIaIaGaGaI...    81   
2   chr13  48689490  tttttttttttttttttttttttttttttttttttttttttttttt...    85   
3   chr13  48689491  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...   141   
4   chr13  48689492  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...   105   
..    ...       ...                                                ...   ...   
82  chr13  48689570                                              aaaaa     5   
83  chr13  48689571                                             cccccc     6   
84  chr13  48689572                                             cccccc     6   
85  chr13  48689573                                             tttttt     6   
86  chr13  48689574                                             cccccc     6   

   Shannon_entropy  
0         1.906454

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


2-2. chrom, pos, pos2를 pileup5로 받고, 계산한 Shannon entropy column을 붙임.

In [ ]:
dpileup5 = dpileup4[['chrom','pos']]
dpileup5['pos2']=dpileup4['pos']+1
dpileup5['Shannon_entropy'] = dpileup4['Shannon_entropy']

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [ ]:
dpileup5

chrom       pos      pos2 Shannon_entropy
0   chr13  48689488  48689489        1.906454
1   chr13  48689489  48689490          1.4998
2   chr13  48689490  48689491        0.988811
3   chr13  48689491  48689492        1.258637
4   chr13  48689492  48689493            -0.0
..    ...       ...       ...             ...
82  chr13  48689570  48689571            -0.0
83  chr13  48689571  48689572            -0.0
84  chr13  48689572  48689573            -0.0
85  chr13  48689573  48689574            -0.0
86  chr13  48689574  48689575            -0.0

[87 rows x 4 columns]

계산한 결과를 Mirlet7d.txt로 빼냄.

In [ ]:
dpileup5.to_csv("Mirlet7d.txt",header=False, index=False, sep='\t')

Bedgraph format의 txt 파일을 만들기 위해 맨 위 row에

track type=bedGraph name="Mirlet7d" description="Mirlet7d" visibility=full color=200,100,0 altColor=0,100,200 priority=20 alwaysZero=on

를 붙이고 저장함.


dpileup5를 dpileup5sorted로 받고, 컬럼 Shannon_entropy를 기준으로 오름차순 정렬.

In [ ]:
dpileup5sorted = dpileup5

In [ ]:
dpileup5sorted.sort_values('Shannon_entropy',ascending=False)

chrom       pos      pos2 Shannon_entropy
0   chr13  48689488  48689489        1.906454
1   chr13  48689489  48689490          1.4998
41  chr13  48689529  48689530        1.450881
39  chr13  48689527  48689528        1.343948
3   chr13  48689491  48689492        1.258637
..    ...       ...       ...             ...
38  chr13  48689526  48689527            -0.0
36  chr13  48689524  48689525            -0.0
35  chr13  48689523  48689524            -0.0
34  chr13  48689522  48689523            -0.0
86  chr13  48689574  48689575            -0.0

[87 rows x 4 columns]

이에 따라 Shannon entropy값이 가장 높은 포지션, 다시 말해 위쪽 row에 있는 base position에서 error가 나올 것으로 추정됨.

다만 visualization 결과 48689488-48689490 부분은 aligment 끝부분으로 보이기 때문에, 48689527-48689530	부분을 unique sequence 확인 target으로 잡음.

해당 부분이 정말 error가 많이 나오는 지 확인해보기 위해, [UCSC Genome Browser](http://genome.ucsc.edu/cgi-bin/hgTracks?db=mm39&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr9%3A106056039-106056126)로 접속 후 custom tracks에 CLIP-let7d_sorted.bam (with CLIP-let7d_sorted.bam.bai)와 Mirlet7d.txt를 submit하고 visualization.

확인 결과, 48689528, 48689529 base에서 unique sequence를 확인함.

View 메뉴의 PDF/PS 를 눌러서 이를 PDF 파일화 (220602_project_2021_36845_Heeun_Kwon_Mirlet7d).

해당 결과를 https://github.com/Heeun93/2021-36845_Homework 에 업로드함.